server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java

```diff
@@ -82,4 +86,15 @@ private ObjectStorageBlobsDAO buildObjectStore(ObjectStorageBlobConfiguration co
         throw new IllegalArgumentException("unknown provider " + configuration.getProvider());
     }
 
+    private Optional<Function<Blob, BlobId>> putBlob(BlobId.Factory blobIdFactory, ObjectStorageBlobConfiguration configuration) {
+        switch (configuration.getProvider()) {
+            case SWIFT:
+                return Optional.empty();
```

Comment:
having a default here would be more appropriate IMO

Reply:
Hard todo without breaking the `Supplier<BlobStore>` mechanism.